File: veltahleon/src/main/java/com/testament/veltahleon/responses/CustomResponse.java
package com.testament.veltahleon.responses;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.http.HttpStatus;

import java.time.LocalDateTime;
import java.util.Map;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CustomResponse {

    private LocalDateTime timestamp;
    private int statusCode;
    private HttpStatus status;
    private Map<?, ?> data;
//    private Map<?, ?> reason;
    private String message;
}

File: interface/frontend/src/app/interfaces/custom-response/custom-response.ts
export interface CustomResponse {
  timestamp: Date;
  statusCode: number;
  status: string;
  data: {
    dataRetrieved?: any[],
    dataSaved?: any,
    dataDeleted?: boolean,
    dataUpdated?: any
  };
  // reason?: {};
  message: string;
}
